handler.h

```diff
@@ -149,4 +150,35 @@ class StatusHandler : public RequestHandler {
 
 REGISTER_REQUEST_HANDLER(StatusHandler);
 
+class ProxyHandler : public RequestHandler {
+public:
+    ProxyHandler() {}
+
+    Status Init(const std::string& uri_prefix,
+                const NginxConfig& config);
+
+    Status HandleRequest(const Request& request,
+                         Response* response);
+
+    std::unique_ptr<Response> get_response(std::string path, std::string host, std::string port);
+
+    size_t get_type(std::string input_string);
```

Comment:
maybe my ctrl-f isnt working but I dont see where the implementation of this function is. What is this for?